configuration.rst

```diff
@@ -186,111 +215,353 @@ This parameter is then referenced in the framework config in
                 https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
 
             <!-- any string surrounded by two % is replaced by that parameter value -->
-            <framework:config default-locale="%locale%">
+            <some-package email-address="%app.admin_email%">
                 <!-- ... -->
-            </framework:config>
+            </some-package>
         </container>
 
     .. code-block:: php
 
-        // config/packages/translation.php
-        $container->loadFromExtension('framework', [
+        // config/packages/some_package.php
+        $container->loadFromExtension('some_package', [
             // any string surrounded by two % is replaced by that parameter value
-            'default_locale' => '%locale%',
+            'email_address' => '%app.admin_email%',
 
             // ...
         ]);
 
-You can define whatever parameter names you want under the ``parameters`` key of
-any configuration file. To reference a parameter, surround its name with two
-percent signs - e.g. ``%locale%``.
+.. include:: /components/dependency_injection/_imports-parameters-note.rst.inc
+
+Configuration parameters are very common in Symfony applications. Some packages
+even define their own parameters (e.g. when installing the translation package,
+a new ``locale`` parameter is added to the ``config/services.yaml`` file).
 
 .. seealso::
 
-    You can also set parameters dynamically, like from environment variables.
-    See :doc:`/configuration/environment_variables`.
+    Read the :ref:`service parameters <service-container-parameters>` article to
+    learn about how to use these configuration parameters in services and controllers.
+
+.. index::
+   single: Environments; Introduction
+
+.. _page-creation-environments:
+.. _page-creation-prod-cache-clear:
+.. _configuration-environments:
+
+Configuration Environments
```

Comment:
I think somewhere in this article, we'll have to explain that all configuration does is actually configuring the services in Service Container. And by configuring the services, all framework behaviour can be changed. This, imo, makes the user understand configuration and environments better (also in combination with service parameters explained above).